core/session.go

```diff
@@ -0,0 +1,154 @@
+package core
+
+import (
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/dicedb/dice/config"
+	"golang.org/x/crypto/bcrypt"
+)
+
+const (
+	DefaultUserName = "default"
+	AuthCmd         = "AUTH"
+
+	SessionStatusPending SessionStatusT = SessionStatusT(0)
+	SessionStatusActive  SessionStatusT = SessionStatusT(1)
+	SessionStatusExpired SessionStatusT = SessionStatusT(2)
+)
+
+var (
+	UserStore *Users = NewUsers()
+)
```

Comment:
Redis must be storing the users in some durable storage, or is everything transient?

Reply:
Prior to Redis 6.0, there was only a concept of a static password which can be defined by passing `requirepass` in the configuration. This PR supports only that.
From Redis 6.0, the concept of ACLs was introduced. User auth storage along with finer grained permissions will be introduced in subsequent PRs.